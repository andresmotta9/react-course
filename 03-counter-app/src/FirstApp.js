// import React, { Fragment } from 'react';
import React from "react";
import PropTypes from "prop-types";

const FirstApp = ({ greet, subtitle }) => {
  // const person =  {
  //     name: 'Andres',
  //     age: 24,
  // }

  return (
    <>
      <h1> {greet} </h1>
      {/* <pre> { JSON.stringify( person, null, 3) } </pre> */}
      <p> {subtitle} </p>
    </>
  );
};

FirstApp.propTypes = {
  greet: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
    subtitle: 'I am a subtitle'
}

export default FirstApp;
