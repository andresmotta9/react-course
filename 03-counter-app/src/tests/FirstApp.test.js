import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import FirstApp from "../FirstApp";
// import { render } from '@testing-library/react';

describe("Tests in <FirstApp />>", () => {
  // test('should show the message -Hi, I am Andres-', () => {
  //     const greet = 'Hi, I am Andres';

  //     const { getByText } = render( <FirstApp greet={ greet }/> );

  //     expect( getByText(greet) ).toBeInTheDocument();
  // });

  test("should show <FirstApp /> correctly", () => {
    const greet = "Hi, I am Andres";
    const wrapper = shallow(<FirstApp greet={greet} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show the subtitle sent by props", () => {
    const greet = "Hi, I am Andres";
    const subtitle = "this is a subtitle";

    const wrapper = shallow(
        <FirstApp
            greet={ greet } 
            subtitle={ subtitle }
        />
    );

    const textParagraph = wrapper.find('p').text();
    expect ( textParagraph ).toBe( subtitle);
  });
});
