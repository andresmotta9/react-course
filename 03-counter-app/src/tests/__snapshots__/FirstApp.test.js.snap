// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp />> should show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
     
    Hi, I am Andres
     
  </h1>
  <p>
    I am a subtitle
  </p>
</Fragment>
`;
